jerry-port/default/CMakeLists.txt

```diff
@@ -26,6 +26,12 @@ file(GLOB SOURCE_PORT_DEFAULT *.c)
 # (should only be necessary if we used compiler default libc but not checking that)
 set(DEFINES_PORT_DEFAULT _BSD_SOURCE _DEFAULT_SOURCE)
 
+INCLUDE (CheckStructHasMember)
+CHECK_STRUCT_HAS_MEMBER ("struct tm" tm_gmtoff time.h HAVE_TM_GMTOFF)
+if(HAVE_TM_GMTOFF)
+  set(DEFINES_PORT_DEFAULT ${DEFINES_PORT_DEFAULT} HAVE_TM_GMTOFF)
+endif()
```

Comment:
I haven't investigated all the details of the problem but my guess is that the root of the problem is that `tm_gmtoff` is non-standard. See, for example, in glibc:
- `tm_gmtoff` is only declared if `__USE_MISC` is defined: https://sourceware.org/git/?p=glibc.git;a=blob;f=time/bits/types/struct_tm.h;h=b13b631228d0ec36691b25db2e1f9b1d66b54bb0;hb=HEAD#l19
- and it only defines `__USE_MISC` if `_DEFAULT_SOURCE` is defined: https://sourceware.org/git/?p=glibc.git;a=blob;f=include/features.h;h=5bed0a499605a3a26d55443f3c8b7e67de152f74;hb=HEAD#l368
- but when a compiler is instructed to compile in strict C99 mode then it will not pass extra macros to the preprocessor and thus will not "enable" such non-standard libc parts.

So, if `CHECK_STRUCT_HAS_MEMBER` calls the compiler with default settings, i.e., with no extra defines, then the compilation during the check will not see `tm_gmtoff`. Perhaps setting `CMAKE_REQUIRED_DEFINITIONS` to `_DEFAULT_SOURCE` before `CHECK_STRUCT_HAS_MEMBER` may solve the problem (see: https://cmake.org/cmake/help/v3.0/module/CheckStructHasMember.html). But even if this works, it may be glibc-specific.

Reply:
I think it is something to do with my defines not getting passed. If I put a message() inside the if, it gets printed (at least, if I run `(cd jerry-port/default && cmake .)`). And compiling outside of cmake or any build system compiles fine without errors:
```c
#include <stdio.h>
#include <time.h>
#include <stdbool.h>

double jerry_port_get_local_time_zone_adjustment (double unix_ms,  /**< ms since unix epoch */
                                                  bool is_utc)  /**< is the time above in UTC? */
{
  struct tm tm;
  time_t now = (time_t) (unix_ms / 1000);
  localtime_r (&now, &tm);
  if (!is_utc)
  {
    now -= tm.tm_gmtoff;
    localtime_r (&now, &tm);
  }
  return ((double) tm.tm_gmtoff) * 1000;
} /* jerry_port_get_local_time_zone_adjustment */

int main() {
	double adj = jerry_port_get_local_time_zone_adjustment(0, 0);
	printf("%lf\n", adj);
}
```
(the above compiles without any warnings or errors with just `gcc test_gmtoff.c`)

I verified that it isn't trying to use `tm.tm_gmtoff` (and getting, say, bogus values) by putting a printf in `default-date.c` before each return statement. The one we hit is the one immediately before `return 0;`, so it's not even attempting to use the `tm.tm_gmtoff` path.

Reply:
Hmm... looking at `CMakeCache.txt` after a clean build, maybe I'm wrong, and for some reason, when run as part of the larger project, `tm.tm_gmtoff` isn't found. 
<img width="247" alt="image" src="https://user-images.githubusercontent.com/667720/46840187-be359480-cd8f-11e8-8936-8a7e411f8da1.png">

Reply:
Confirmed that when running `cmake .` in `jerry-port/default`, my `CMakeCache.txt` is generated with `HAVE_TM_GMTOFF:INTERNAL=1`, but when run as part of `tools/build.py`, the generated `CMakeCache.txt` (in `build/tests/test262_tests/CMakeCache.txt`) has `HAVE_TM_GMTOFF:INTERNAL=`, as above.

Looking at the two different `CMakeCache.txt` files, no glaring differences in build configuration stick out at me.

Reply:
Ah, I see now what you were saying. I didn't realize jerryscript compiled in strict c99 mode (which, as you said, makes `tm.tm_gmtoff` invisible). `gcc -std=c99 test_gmtoff.c` does indeed fail as you predicted.

Reply:
Huh, so the default source code that `check_struct_has_member` uses to check if the given struct has the given member actually _fails to compile_ with the default jerryscript compiler options... Ugh.
```
Performing C SOURCE FILE Test HAVE_TM_GMTOFF failed with the following output:
Change Dir: /home/fbuser/src/jerryscript/build/tests/test262_tests/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_a80df/fast"
/usr/bin/make -f CMakeFiles/cmTC_a80df.dir/build.make CMakeFiles/cmTC_a80df.dir/build
make[1]: Entering directory '/home/fbuser/src/jerryscript/build/tests/test262_tests/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_a80df.dir/src.c.o
/usr/bin/cc  -D_DEFAULT_SOURCE  -flto -fno-fat-lto-objects -std=c99 -pedantic -fno-builtin -fno-stack-protector -Wall -Werror=all -Wextra -Werror=extra -Wformat-nonliteral -Werror=format-nonliteral -Winit-self -Werror=init-self -Wconversion -Werror=conversion -Wsign-conversion -Werror=sign-conversion -Wformat-security -Werror=format-security -Wmissing-declarations -Werror=missing-declarations -Wshadow -Werror=shadow -Wstrict-prototypes -Werror=strict-prototypes -Wundef -Werror=undef -Wold-style-definition -Werror=old-style-definition -Wno-stack-protector -Wno-attributes -Werror -Wlogical-op -Werror=logical-op -DHAVE_TM_GMTOFF   -o CMakeFiles/cmTC_a80df.dir/src.c.o   -c /home/fbuser/src/jerryscript/build/tests/test262_tests/CMakeFiles/CMakeTmp/src.c
/home/fbuser/src/jerryscript/build/tests/test262_tests/CMakeFiles/CMakeTmp/src.c:4:5: error: function declaration isn’t a prototype [-Werror=strict-prototypes]
 int main()
     ^
/home/fbuser/src/jerryscript/build/tests/test262_tests/CMakeFiles/CMakeTmp/src.c: In function ‘main’:
/home/fbuser/src/jerryscript/build/tests/test262_tests/CMakeFiles/CMakeTmp/src.c:4:5: error: old-style function definition [-Werror=old-style-definition]
cc1: all warnings being treated as errors
CMakeFiles/cmTC_a80df.dir/build.make:65: recipe for target 'CMakeFiles/cmTC_a80df.dir/src.c.o' failed
make[1]: *** [CMakeFiles/cmTC_a80df.dir/src.c.o] Error 1
make[1]: Leaving directory '/home/fbuser/src/jerryscript/build/tests/test262_tests/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_a80df/fast' failed
make: *** [cmTC_a80df/fast] Error 2

Source file was:

#include <time.h>

int main()
{
   (void)sizeof(((struct tm *)0)->tm_gmtoff);
   return 0;
}
```

Reply:
Finally got it all to build. On the plus side, I'm a lot more familiar with cmake now :).